src/forms/userForm.tsx

```diff
@@ -0,0 +1,372 @@
+                  onChange={() => {
+                    this.setState({
+                      permissions: this.updateBooleanPermissions(
+                        "power_control"
+                      )
+                    });
+                  }}
+                ></Checkbox>
+              </FormGroup>
+              <FormGroup inline={true}>
+                <Checkbox
+                  label="Audit Log"
+                  checked={
+                    this.state.permissions.admin
+                      ? true
+                      : this.state.permissions.audit_read
+                  }
+                  alignIndicator={Alignment.LEFT}
+                  disabled={this.state.permissions.admin}
+                  onChange={() => {
+                    this.setState({
+                      permissions: this.updateBooleanPermissions("audit_read")
+                    });
+                  }}
+                ></Checkbox>
+              </FormGroup>
+              <RadioGroup
+                inline={true}
+                label="Datacenter permissions"
+                onChange={() => {
+                  console.log("changing datacenter");
+                  console.log(this.state);
+                  var updatedPermissions = this.state.permissions;
+                  if (this.state.datacenter_selection === "Global") {
+                    let permissions: Array<string>;
+                    permissions = [];
+                    for (var i = 0; i < this.state.datacenters.length; i++) {
+                      permissions.push(this.state.datacenters[i].id);
+                    }
+                    updatedPermissions.datacenter_permissions = permissions;
+                    this.setState({
+                      datacenter_selection: "Per Datacenter",
+                      permissions: updatedPermissions
+                    });
+                  } else {
+                    updatedPermissions.datacenter_permissions = this.state.initialValues.datacenter_permissions;
+                    this.setState({
+                      datacenter_selection: "Global",
+                      permissions: updatedPermissions
+                    });
+                  }
+                }}
+                selectedValue={this.state.datacenter_selection}
+              >
+                <Radio label="Global" value="Global" />
+                <Radio label="Per Datacenter" value="Per Datacenter" />
+              </RadioGroup>
+              {this.state.datacenter_selection === "Global"
+                ? null
+                : this.renderDatacenterChecks()}
+              <Button className="login-button" type="submit">
+                Submit
+              </Button>
+            </form>
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  private renderDatacenterChecks() {
+    var checks: Array<any>;
+    checks = [];
+    for (var i = 0; i < this.state.datacenters.length; i++) {
```

Comment:
could we sort these alphabetically so it's easier for the user?? if that's a lot of extra work then don't worry about it